### ISSUE TITLE
feat: Add missing issue transition tools (confirmIssue, resolveIssue, reopenIssue)

### ISSUE THREAD

**Issue by sapientpants** [sapientpants/sonarqube-mcp-server#135]
## Summary

Add new MCP tools for missing SonarQube issue transitions following the existing `markIssueXXX` naming pattern. This will provide complete coverage of SonarQube's issue transition capabilities through dedicated, well-named tools.

## Background

Currently, we have:
- `markIssueFalsePositive` - Transitions issue to false positive
- `markIssueWontFix` - Transitions issue to won't fix

SonarQube supports additional transitions that should be exposed:
- `confirm` - Confirm an issue
- `unconfirm` - Unconfirm an issue  
- `reopen` - Reopen a resolved issue
- `resolve` - Resolve an issue

#### 1. `confirmIssue` Tool

**Types (`src/types/issues.ts`):**
```typescript
export interface ConfirmIssueParams {
  issueKey: string;
  comment?: string;
}
```

**Schema (`src/schemas/issues.ts`):**
```typescript
export const confirmIssueToolSchema = {
  issue_key: z.string().describe('The key of the issue to confirm'),
  comment: z.string().optional().describe('Optional comment explaining why this issue is confirmed'),
};
```

**Handler (`src/handlers/issues.ts`):**
```typescript
export async function handleConfirmIssue(
  params: ConfirmIssueParams,
  client: ISonarQubeClient = getDefaultClient()
) {
  logger.debug('Handling confirm issue request', { issueKey: params.issueKey });

  try {
    const result = await client.confirmIssue(params);
    logger.info('Successfully confirmed issue', {
      issueKey: params.issueKey,
      comment: params.comment ? 'with comment' : 'without comment',
    });

    return {
      content: [
        {
          type: 'text' as const,
          text: JSON.stringify({
            message: `Issue ${params.issueKey} confirmed`,
            issue: result.issue,
            components: result.components,
            rules: result.rules,
            users: result.users,
          }),
        },
      ],
    };
  } catch (error) {
    logger.error('Failed to confirm issue', error);
    throw error;
  }
}
```

#### 2. `unconfirmIssue` Tool

Similar pattern with `UnconfirmIssueParams`, `unconfirmIssueToolSchema`, and `handleUnconfirmIssue`.

#### 3. `resolveIssue` Tool

**Schema (`src/schemas/issues.ts`):**
```typescript
export const resolveIssueToolSchema = {
  issue_key: z.string().describe('The key of the issue to resolve'),
  comment: z.string().optional().describe('Optional comment explaining how the issue was resolved'),
};
```

#### 4. `reopenIssue` Tool

**Schema (`src/schemas/issues.ts`):**
```typescript
export const reopenIssueToolSchema = {
  issue_key: z.string().describe('The key of the issue to reopen'),
  comment: z.string().optional().describe('Optional comment explaining why the issue is being reopened'),
};
```

### Domain Methods (`src/domains/issues.ts`)

Add these methods following the existing pattern:

```typescript
async confirmIssue(params: ConfirmIssueParams): Promise<DoTransitionResponse> {
  const request = {
    issue: params.issueKey,
    transition: 'confirm' as const,
  };

  if (params.comment) {
    await this.webApiClient.issues.addComment({
      issue: params.issueKey,
      text: params.comment,
    });
  }

  return this.webApiClient.issues.doTransition(request);
}

async unconfirmIssue(params: UnconfirmIssueParams): Promise<DoTransitionResponse> {
  const request = {
    issue: params.issueKey,
    transition: 'unconfirm' as const,
  };

  if (params.comment) {
    await this.webApiClient.issues.addComment({
      issue: params.issueKey,
      text: params.comment,
    });
  }

  return this.webApiClient.issues.doTransition(request);
}

async resolveIssue(params: ResolveIssueParams): Promise<DoTransitionResponse> {
  const request = {
    issue: params.issueKey,
    transition: 'resolve' as const,
  };

  if (params.comment) {
    await this.webApiClient.issues.addComment({
      issue: params.issueKey,
      text: params.comment,
    });
  }

  return this.webApiClient.issues.doTransition(request);
}

async reopenIssue(params: ReopenIssueParams): Promise<DoTransitionResponse> {
  const request = {
    issue: params.issueKey,
    transition: 'reopen' as const,
  };

  if (params.comment) {
    await this.webApiClient.issues.addComment({
      issue: params.issueKey,
      text: params.comment,
    });
  }

  return this.webApiClient.issues.doTransition(request);
}
```

### Client Methods (`src/sonarqube.ts`)

Add after the existing transition methods:

```typescript
async confirmIssue(params: ConfirmIssueParams): Promise<DoTransitionResponse> {
  return this.issuesDomain.confirmIssue(params);
}

async unconfirmIssue(params: UnconfirmIssueParams): Promise<DoTransitionResponse> {
  return this.issuesDomain.unconfirmIssue(params);
}

async resolveIssue(params: ResolveIssueParams): Promise<DoTransitionResponse> {
  return this.issuesDomain.resolveIssue(params);
}

async reopenIssue(params: ReopenIssueParams): Promise<DoTransitionResponse> {
  return this.issuesDomain.reopenIssue(params);
}
```

### Tool Registration (`src/index.ts`)

Register after existing issue tools:

```typescript
// Confirm issue
export const confirmIssueHandler = async (params: Record<string, unknown>) => {
  return handleConfirmIssue({
    issueKey: params.issue_key as string,
    comment: params.comment as string | undefined,
  });
};

mcpServer.tool(
  'confirmIssue',
  'Confirm a SonarQube issue',
  confirmIssueToolSchema,
  createToolHandler(confirmIssueHandler)
);

// Similar registration for unconfirmIssue, resolveIssue, and reopenIssue
```

## Benefits

1. **Consistency**: Follows the established `markIssueXXX` pattern
2. **Clarity**: Each tool has a clear, specific purpose
3. **Discoverability**: Tools are easy to find and understand
4. **Type Safety**: Each transition has its own types and validation

## Example Usage

```typescript
// Confirm an issue
await mcp.callTool('confirmIssue', {
  issue_key: 'PROJECT-123',
  comment: 'Confirmed after code review'
});

// Resolve an issue
await mcp.callTool('resolveIssue', {
  issue_key: 'PROJECT-456',
  comment: 'Fixed in commit abc123'
});

// Reopen an issue
await mcp.callTool('reopenIssue', {
  issue_key: 'PROJECT-789',
  comment: 'Issue still occurs in production'
});

// Unconfirm an issue
await mcp.callTool('unconfirmIssue', {
  issue_key: 'PROJECT-012',
  comment: 'Needs further investigation'
});
```

## Additional Considerations

1. **Error Handling**: Each tool should handle invalid transitions gracefully
2. **Documentation**: Update README with all available issue transition tools
3. **Testing**: Add comprehensive tests for each new tool
4. **Permissions**: Note that some transitions may require specific permissions in SonarQube

## References

- SonarQube Web API documentation: `/api/issues/do_transition`
- Existing implementations: `markIssueFalsePositive` and `markIssueWontFix` in `src/domains/issues.ts`
